### PR TITLE
fix(TPG > 4.73)!: Ignore maintenance_schedule plan changes

### PR DESCRIPTION
**Pull request by gservat**
Propose we remove this lifecycle setting as I do not believe it is required anymore as `maintenance_schedule` is output only.

Related issue: https://github.com/terraform-google-modules/terraform-google-memorystore/issues/166
Reverts terraform-google-modules/terraform-google-memorystore#152

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -66,10 +66,6 @@ resource "google_redis_instance" "default" {
       rdb_snapshot_period = persistence_config.value["rdb_snapshot_period"]
     }
   }
-
-  lifecycle {
-    ignore_changes = [maintenance_schedule]
```
Comment (dubb-b) [inline main.tf:71]: Should it just be ` [maintenance_policy]` ??

Comment (gservat) [reply main.tf:71]: @dubb-b not sure I follow. It's currently set to `maintenance_policy`?

https://github.com/terraform-google-modules/terraform-google-memorystore/blob/531de8fe9218900117be02117982926ae50f25bb/main.tf#L70-L72

Comment (gservat) [reply main.tf:71]: Oh I see what you mean. This PR is about reverting the lifecycle policy as `maintenance_schedule` is now correctly marked as output only (fixed in https://github.com/GoogleCloudPlatform/magic-modules/pull/8261).

**Review by apeabody — CHANGES_REQUESTED** [review]
Thanks for the contribution @gservat!

Can you please bump to minimum provider to the version with this fix? https://github.com/terraform-google-modules/terraform-google-memorystore/blob/master/versions.tf#L23

We'll also want to make this a `fix!:` PR as the provider version will be a breaking change.

**Comment by imrannayer** [comment]
@gservat r u still working on this PR?

**Comment by imrannayer** [comment]
Fixed in #195

**Pull request closed without merging**